lib/barbeque/message_handler/job_execution.rb

```diff
@@ -6,19 +6,21 @@ module Barbeque
   module MessageHandler
     class JobExecution
       # @param [Barbeque::Message::JobExecution] message
-      # @param [Barbeque::JobQueue] job_queue
-      def initialize(message:, job_queue:)
+      # @param [Barbeque::MessageQueue] message_queue
+      def initialize(message:, message_queue:)
         @message = message
-        @job_queue = job_queue
+        @message_queue = message_queue
       end
 
       def run
         begin
-          job_execution = Barbeque::JobExecution.create(message_id: @message.id, job_definition: job_definition, job_queue: @job_queue)
+          job_execution = Barbeque::JobExecution.create(message_id: @message.id, job_definition: job_definition, job_queue: @message_queue.job_queue)
         rescue ActiveRecord::RecordNotUnique => e
+          @message_queue.delete_message(@message)
```

Comment:
> That is, can we omit DeleteMessage when DuplicatedExecution happens?

I think so in normal (successful dequeue worker does not shut down unexpectedly) situation.

But if it shuts down unexpectedly before DeleteMessage, the message would be unnecessarily dequeued multiple times. So it might be better to delete the message here checking it's already stored to S3 (and store it if it's not).